src/locale.js

```diff
@@ -146,3 +184,85 @@ export default class Locale {
 function getLanguageDirection( languageCode ) {
 	return RTL_LANGUAGE_CODES.includes( languageCode ) ? 'rtl' : 'ltr';
 }
+
+/**
+ * A translated message returned from one of the {@link Locale}'s methods.
+ */
+export class Message {
+	/**
+	 * @hidden
+	 * @param {String} message
+	 */
+	constructor( message ) {
+		/**
+		 * @readonly
+		 * @member {String}
+		 */
+		this.message = String( message );
+	}
+
+	/**
+	 * Get the message as a string without any transformations
+	 * @returns {String}
+	 */
+	toString() {
+		return this.message;
+	}
+
+	/**
+	 * Get the interpolated message as string.
+	 * This method will replace the placeholders in the message with provided values.
+	 *
+	 * @example
+	 *
+	 * 		const msg = ct( 'Replace text', 'Replace %0 with %1' )
+	 * 			.format( 'foo', 'bar' )
+	 *
+	 * 		expect( msg ).to.equal( 'Replace foo with bar' )
+	 *
+	 * @param {...String} values
+	 * @returns {String} The interpolated message
+	 */
+	format( ...values ) {
+		return this.message.replace( /%(\d+)/g, ( match, index ) => values[ parseInt( index, 10 ) ] );
+	}
+
+	/**
+	 * Get the interpolated message as an array of static and dynamic parts.
+	 * This method splits the message at every placeholder
+	 * and returns an array of the original string parts with provided values interspersed.
+	 * The values can be of any type and won't be stringified.
+	 *
+	 * @example
+	 *
+	 * 		const msg = ct( 'Replace text', 'Replace %0 with %1' )
+	 * 			.formatToParts( 'foo', 'bar' )
+	 *
+	 * 		expect( msg ).to.equal( [ 'Replace ', 'foo', ' with ', 'bar' ] )
+	 *
+	 * @template T
```

Comment:
I'm also unsure about the generic `@template` Currently we don't use it anywhere. I know it's a very valuable thing, though it can be unsupported currently by Uberto and our JSDoc plugins 😞